### ISSUE TITLE
There is no DbProviderFactory implementation.

### ISSUE THREAD

**Issue by jeremychild** [bgrainger/MySqlConnector#34]
Hi. Great work with the project! I have found that there is no DbProviderFactory implementation. I feel this would be useful integrating with ORM's and other existing libraries that require a System.Data.Common.DbProviderFactory.

To get the ball rolling I suggest we copy how [Microsoft.Data.Sqlite](https://github.com/aspnet/Microsoft.Data.Sqlite) have done theirs. [SqliteFactory.cs](https://github.com/aspnet/Microsoft.Data.Sqlite/blob/dev/src/Microsoft.Data.Sqlite/SqliteFactory.cs)

An example would be

```
public class MySqlClientFactory : DbProviderFactory
{
    public static readonly MySqlClientFactory Instance = new MySqlClientFactory();

    private MySqlClientFactory()
    {
    }

    public override DbCommand CreateCommand()
        => new MySqlCommand();

    public override DbConnection CreateConnection()
        => new MySqlConnection();

    public override DbConnectionStringBuilder CreateConnectionStringBuilder()
        => new MySqlConnectionStringBuilder();

    public override DbParameter CreateParameter()
        => new MySqlParameter();
}
```

**Comment by jeremychild** [comment]
I have marked the class as `sealed` and submitted an initial PR. See pull #35

**Comment by bgrainger** [comment]
Shipped in [0.1.0-alpha13](https://www.nuget.org/packages/MySqlConnector/0.1.0-alpha13).
